### ISSUE TITLE
[Task]: Update BigQueryIO.setTriggeringFrequency documentation

### ISSUE THREAD

**Issue by johnjcasey** [apache/beam#23949]
### What needs to happen?

BigQueryIO currently documents triggering frequency as:

This is only applicable when the write method is set to {@link Method#FILE_LOADS}, and only when writing an unbounded {@link PCollection}.

However, we use triggering frequency for STORAGE_WRITE_API and STORAGE_API_AT_LEAST_ONCE as well. This documentation should be updated to reflect this.

In addition, during the validation component of BigQueryIO.write().expand(), we check for unbounded && (file loads || storage write api) to see if a triggering frequency is set. This should probably also be checked if storage write at least once is set. 

### Issue Priority

Priority: 2

### Issue Component

Component: io-java-gcp

**Comment by DhirajSardar** [comment]
I would like to contribute . Can anyone help me ? I am a newbie.

**Comment by nikhilpothuru** [comment]
.take-issue

**Comment by nikhilpothuru** [comment]
@johnjcasey Hi John, I was reading through the guidelines for joining the ASF slack workspace and it looks like non-ASF members (those without an @apache.org address) need to be invited. I was wondering if I could join the workspace. My email is as follows: pothurunikhil@gmail.com

**Comment by Vivek-2004** [comment]
I want to work on this issue .

**Comment by johnjcasey** [comment]
> pothurunikhil@gmail.com

Sorry it took a while, you're invited

**Comment by johnjcasey** [comment]
@Vivek-2004 Sounds good! Let me know how I can help

**Comment by Vivek-2004** [comment]
I guess I have a solution to this issue... Would try to implement the solution once the issue is assigned to me

**Comment by johnjcasey** [comment]
you can assign it to yourself using the command ".take-issue"

**Comment by 0ChanAk** [comment]
.take-issue

**Comment by Ayush991rec** [comment]
.take-issue

**Comment by rajkgupt** [comment]
.take-issue

**Comment by Arunodoy18** [comment]
hi, I can solve this issue , i would like to contribute.
.take-issue

**Comment by tarun-google** [comment]
I was going through BigQueryIO. looks like the triggering frequency is actually ignored when the method is STORAGE_API_AT_LEAST_ONCE. So, i do not see any issue

1. The validation throws the warning that triggering frequency is ignored https://github.com/apache/beam/blob/31a96b6807ae22765c76ef2184ace106e27ed154/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L3762
2. Inconsistent writes doe not create groups based on triggering frequency while 
https://github.com/apache/beam/blob/31a96b6807ae22765c76ef2184ace106e27ed154/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiLoads.java#L158

**Comment by Vicky-8223** [comment]
I would like to contribute on this issue.I am new contributer and would like to take this up.
.take-issue

**Comment by shivanshjais22** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by KartikeyTripathi469** [comment]
Hello, I would like to work on this issue.
Could you please assign it to me?

**Comment by Subramanya-Veeregowda** [comment]
Hi, I noticed there is already an open PR (#37205) addressing this issue.

I went through the discussion and would like to help if there are any pending changes or gaps that still need to be addressed. Please let me know if I can contribute.

Thanks!

**Comment by Subramanya-Veeregowda** [comment]
Hi, I noticed this issue and PR(#37205 ) has been inactive for a while and has some conflicts.

I’d like to help move this forward. I will open a new one with a minimal fix (only updating the relevant Javadoc)?

@ahmedabu98 if there is any specific requirements then mention it ready to work on it.

**Comment by ahmedabu98** [comment]
> I’d like to help move this forward. I will open a new one with a minimal fix (only updating the relevant Javadoc)?

Hey @Subramanya-Veeregowda, yes that should be enough to get it in!

**Comment by damccorm** [comment]
> I was going through BigQueryIO. looks like the triggering frequency is actually ignored when the method is STORAGE_API_AT_LEAST_ONCE. So, i do not see any issue
> 
> 1. The validation throws the warning that triggering frequency is ignored https://github.com/apache/beam/blob/31a96b6807ae22765c76ef2184ace106e27ed154/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L3762
> 2. Inconsistent writes doe not create groups based on triggering frequency while
>    https://github.com/apache/beam/blob/31a96b6807ae22765c76ef2184ace106e27ed154/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiLoads.java#L158

@tarun-google should the issue be closed then?

**Comment by tarun-google** [comment]
Yes, we can close this issue. I asked AI to cross validate since its been a while

Notes:
1. STORAGE_API_AT_LEAST_ONCE intentionally does not use triggeringFrequency—it routes to expandInconsistent() which streams writes directly without batching or buffering delay, and BigQueryIO.expand() explicitly disallows/ignores triggering frequency for this method.
2. The Javadoc update for STORAGE_WRITE_API was already merged in #30457 (which clarified that withTriggeringFrequency applies to FILE_LOADS and STORAGE_WRITE_API).
